lib/mqtt_abstraction_impl.cpp

```diff
@@ -200,12 +211,18 @@ std::future<void> MQTTAbstractionImpl::spawn_main_loop_thread() {
             while (this->mqtt_is_connected) {
 
                 eventfd_t eventfd_buffer;
-                struct pollfd pollfds[2] = {{this->mqtt_socket_fd, POLLIN, 0}, {this->event_fd, POLLIN, 0}};
-                auto retval = ::poll(pollfds, 2, mqtt_poll_timeout_ms);
+                const int nfds = 3;
+                struct pollfd pollfds[nfds] = {{this->mqtt_socket_fd, POLLIN, 0},
+                                               {this->event_fd, POLLIN, 0},
+                                               {this->disconnect_event_fd, POLLIN, 0}};
+                auto retval = ::poll(pollfds, nfds, mqtt_poll_timeout_ms);
 
                 if (retval >= 0) {
                     // data available to send (the notifier writes, we should be ready to read)
                     if (retval > 0) {
+                        if (pollfds[2].revents & POLLIN) {
```

Comment:
Can you add a comment that explains that we're disconnecting here?

Reply:
Done